lib/game.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  attr_reader :score, :scorecard
+
+  def initialize(frame1 = Frame.new, frame2 = Frame.new,
+                 frame3 = Frame.new, frame4 = Frame.new,
+                 frame5 = Frame.new, frame6 = Frame.new,
+                 frame7 = Frame.new, frame8 = Frame.new,
+                 frame9 = Frame.new, frame10 = Frame.new)
+    @score = 0
+    @scorecard = { frame1: frame1, frame2: frame2,
+                   frame3: frame3, frame4: frame4,
+                   frame5: frame5, frame6: frame6,
+                   frame7: frame7, frame8: frame8,
+                   frame9: frame9, frame10: frame10 }
+  end
```

Comment:
I actually kind of like this - it's very clear what it's doing and that the game will be 10 frames.

If you wanted to refactor like you were saying, you could initialize frames on the fly instead, but basically would do the same job so just a matter of preference and design I guess